syntax = "proto3";

package v3;

option go_package = "github.com/dendrascience/dendra-data-api/release/go/v3";

import "v3/types.proto";
import "google/protobuf/empty.proto";

service MetadataService {
  rpc GetDatastream (GetDatastreamRequest) returns (GetDatastreamResponse) {}
  rpc ListUoms (google.protobuf.Empty) returns (ListUomsResponse) {}
}

message GetDatastreamRequest {
  // TODO: add auth token!

  string datastream_id = 1;
}

message GetDatastreamResponse {
  Datastream datastream = 1;
}

message ListUomsResponse {
  repeated Uom uoms = 1 [json_name="data"];
}
